src/monster860/fastdmm/FastDMM.java

```diff
@@ -482,27 +482,33 @@ else if(dwheel < 0)
 		
 		while(Mouse.next()) {
 			if(Mouse.getEventButtonState()) {
+				try {
+					currPopup.setVisible(false);
+				} catch (NullPointerException e) {
+					// Doesn't exist. Oh well.
+				}
 				Location l = new Location(selX, selY, 1);
 				String key = dmm.map.get(l);
 				if(Mouse.getEventButton() == 1) {
 					if(key != null) {
 						TileInstance tInstance = dmm.instances.get(key);
-						currPopup = new PopupMenu();
+						currPopup = new JPopupMenu();
+						currPopup.setLightWeightPopupEnabled(false);
 						for(ObjInstance i : tInstance.getLayerSorted()) {
 							if(i == null)
 								continue;
-							Menu menu = new Menu(i.typeString());
+							JMenu menu = new JMenu(i.typeString());
 							currPopup.add(menu);
 							
-							MenuItem item = new MenuItem("Delete");
+							JMenuItem item = new JMenuItem("Delete");
 							item.addActionListener(new DeleteListener(this, l, i));
 							menu.add(item);
 							
-							item = new MenuItem("View Variables");
+							item = new JMenuItem("View Variables");
 							item.addActionListener(new EditVarsListener(this, l, i));
 							menu.add(item);
 						}
-						canvas.add(currPopup);
+						canvas.getParent().add(currPopup);
```

Comment:
Have you tested this to make sure the right click menu actually shows up in the correct place?